spec/system/header_spec.rb

```diff
@@ -189,23 +196,37 @@
     end
   end
 
-  private
-
-  def get_computed_style_value(selector, property)
-    page.evaluate_script(
-      "window.getComputedStyle(document.querySelector('#{selector}')).getPropertyValue('#{property}')",
-    ).strip
-  end
-
-  def resize_element(selector, size)
-    page.evaluate_script("document.querySelector('#{selector}').style.height = '#{size}px'")
-  end
-
-  def active_element_id
-    page.evaluate_script("document.activeElement.id")
-  end
+  context "when cmd + f keyboard shortcut pressed" do
+    context "when within a topic with less than 20 posts" do
+      fab!(:post) { Fabricate(:post, topic: topic) }
+      it "does not open search" do
+        sign_in(current_user)
+        visit "/t/#{topic.slug}/#{topic.id}"
+        header.search_in_topic_keyboard_shortcut
+        expect(search).not_to have_search_menu_visible
+      end
+    end
 
-  def click_outside
-    find(".d-modal").click(x: 0, y: 0)
+    context "when within a topic with 20+ posts" do
+      before do
+        TopicView.stubs(:chunk_size).returns(2)
+        sign_in(current_user)
+      end
+
+      fab!(:posts) { Fabricate.times(3, :post, topic: topic) }
+
+      it "opens search" do
+        visit "/t/#{topic.slug}/#{topic.id}"
+        header.search_in_topic_keyboard_shortcut
+        expect(search).to have_search_menu_visible
+      end
+
+      it "closes search after two presses" do
+        visit "/t/#{topic.slug}/#{topic.id}"
+        header.search_in_topic_keyboard_shortcut
+        header.search_in_topic_keyboard_shortcut
+        expect(search).not_to have_search_menu_visible
+      end
```

Comment:
Booting the app in system specs is quite costly, so IMO we should combine these two specs. Something like

```
it "opens search on first ctrl+f, then closes on second"
```